aten/src/ATen/cuda/Atomic.cuh

```diff
@@ -402,6 +430,29 @@ __host__ __device__ T safe_max(T a, T b) {
   return max;
 }
 
+__inline__ __device__ c10::complex<float> complex_max(c10::complex<float> a, c10::complex<float> b) {
+  if(at::_isnan(b)) {
```

Comment:
Inlined to avoid multiple definition errors.

Using the magnitude to order the complex numbers(since they are not straight forward to order). I am using intrinsic functions from CUDA but please let me know if these functions should also be callable from host code - if so, I can resort to regular instructions.